File: selenium_junit5_parallel_with_wdm/src/test/java/org/o2f/pages/automationpractice/SearchResultPage.java
package org.o2f.pages.automationpractice;

import org.junit.jupiter.api.Assertions;
import org.o2f.pages.BasePage;
import org.o2f.pages.automationpractice.components.HeaderComponent;
import org.o2f.pages.automationpractice.components.ResultsComponent;
import org.openqa.selenium.WebDriver;

public class SearchResultPage extends BasePage {

    private HeaderComponent headerComponent;
    private ResultsComponent resultsComponent;

    public SearchResultPage(WebDriver driver) {
        super(driver);
        this.headerComponent = new HeaderComponent(driver);
        this.resultsComponent = new ResultsComponent(driver);
    }

    public void isLoaded() {
        resultsComponent.isLoaded();
        //... in a normal automation would have the remaining components
    }

    public void displayedSearchTermsShouldBe(String text) {
        Assertions.assertEquals(text, resultsComponent.getSearchedTermText());
    }

    public void numberOfResultsShouldBe(int ExpectedNumberOfResults) {
        Assertions.assertEquals(ExpectedNumberOfResults, resultsComponent.getNumberOfResults());
    }


}

File: python-selenium-pure/pages/automation_practice/search_result_page.py
from pages.automation_practice.components.header_component import HeaderComponent
from pages.automation_practice.components.results_component import ResultsComponent
from pages.base_page import BasePage


class SearchResultPage(BasePage):

    def __init__(self, driver):
        super().__init__(driver)
        self.header_component = HeaderComponent(driver)
        self.result_component = ResultsComponent(driver)

    def is_loaded(self):
        self.result_component.is_loaded()
        # in a normal automation should have the remaining components

    def displayed_search_term_should_be(self, text: str):
        assert text == self.result_component.get_searched_term()

    def nr_of_result_should_be(self, nr: int):
        assert nr == self.result_component.get_nr_of_results()